fitbenchmarking/parsing/nist_parser.py

```diff
@@ -89,6 +100,30 @@ def _parse_jacobian(self, name):
         jac = jac.split("=")[1].replace("{", "").replace("}", "")
         return jac.split(",")
 
+    def _parse_hessian(self, name):
+        """
+        Parses the Hessian for the NIST file format
+
+        :param name: name of the NIST file
+        :type name: str
+        """
+        file_dir = os.path.abspath(os.path.join(self._filename, os.pardir))
+        hes_file = os.path.join(file_dir, "data_files", "{}.hes".format(name))
+        try:
+            hes_data = open(hes_file, "r")
```

Comment:
This file needs to be closed. Generally this is done using the with syntax, where exiting the indented block cleans up the files for you:
```
with open(...) as f:
    lines = f.readlines()

...
```

Reply:
This is the same for the `_parse_jacobian` function - I can either fix both in this PR or the issue can be addressed in #901, what do you think?

Reply:
I think we should change this one as it's not just a style/implementation change that's required. Although I don't have a strong preference over using the context manager (with) vs just closing explicitly (`f.flush();f.close()`) if it's temporary -- after #901 it should be a with statement though.